### DISCUSSION TITLE
[general] Unnessicary targets build for dynamic branching in combination with later subsetting

### DISCUSSION THREAD

**Discussion by bart1** [ropensci/targets#1397] [General]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

For some project I was exploring using the `slice` feature. The actual case is more elaborate containing nested `tar_map` but the example below should illustrate. I basically want to generate diagnostic plots (here `tt`) for a subset of the dynamic targets before committing to building all (here `y`). I thought the script below would do that. However all (five) branches of `y` are build while only two are needed. Is there a way to prevent that from happening?

``` r
targets::tar_dir({ 
  targets::tar_script({
    list(
        targets::tar_target(x, 1:5),
        targets::tar_target(y, x , pattern = map(x)),
        targets::tar_target(tt, y/2, pattern = slice(y, index=c(2,4)))
    )
  })
  targets::tar_manifest()
  targets::tar_make( tidyselect::contains("tt") )
  targets::tar_read(tt)
})
#> ▶ dispatched target x
#> ● completed target x [0 seconds, 98 bytes]
#> ▶ dispatched branch y_5015dfa5778ec11d
#> ● completed branch y_5015dfa5778ec11d [0 seconds, 47 bytes]
#> ▶ dispatched branch y_5ea9d15a285adf66
#> ● completed branch y_5ea9d15a285adf66 [0.001 seconds, 48 bytes]
#> ▶ dispatched branch y_20a2bb14d64371f8
#> ● completed branch y_20a2bb14d64371f8 [0 seconds, 47 bytes]
#> ▶ dispatched branch y_4cd96100022a8ff3
#> ● completed branch y_4cd96100022a8ff3 [0 seconds, 48 bytes]
#> ▶ dispatched branch y_4540b7c6f3594628
#> ● completed branch y_4540b7c6f3594628 [0 seconds, 47 bytes]
#> ● completed pattern y 
#> ▶ dispatched branch tt_9e1caabb38b0f599
#> ● completed branch tt_9e1caabb38b0f599 [0 seconds, 50 bytes]
#> ▶ dispatched branch tt_0b133e4421ecb8d4
#> ● completed branch tt_0b133e4421ecb8d4 [0 seconds, 49 bytes]
#> ● completed pattern tt 
#> ▶ ended pipeline [0.138 seconds]
#> tt_9e1caabb38b0f599 tt_0b133e4421ecb8d4 
#>                   1                   2
```

<sup>Created on 2024-12-12 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by wlandau** [comment]
To only build the analogous subset of branches of `y`, you could try `targets::tar_target(y, x , pattern = slice(x, index = c(2, 4)))`

**Reply by bart1** [reply]
Thank you for the quick answer! That is a bit less convenient as I need to remove that `slice` if I want to build the full set of `y` values later for final analysis but does work in a quick test
